### PR TITLE
Add doc for Tokenlon

### PR DESCRIPTION
**Pull request by charlesjhongc**

### CODE REVIEW

File: doc/README.md
Comment (pilagod) [inline doc/README.md:36]: nit: Capitalize `peripherals` 😁

**Review by pilagod — COMMENTED** [review]
The structure LGTM 👍

File: doc/README.md
Comment (doublespending) [inline doc/README.md:19]: Strategy contracts are not mentioned later in the article? And
shall we explain the concept of strategy contract?

File: doc/README.md
Comment (doublespending) [inline doc/README.md:42]: Shall we take L2Deposit apart from Executing Contract, list it under `L2 Related Contracts` or `Miscellaneous Contracts` and finally rename `Executing Contracts` to `Trading Contracts` or `Strategy Contracts`?

File: doc/README.md
Comment (changwu-tw) [reply doc/README.md]: how about having the structure like this? 

```
/docs 
├── README.md
├── advanced
│   └── universal_permit.md
├── architecture
│   └── protocol.md
├── peripheral
│   └── LON.md
└── strategy
    └── RFQ.md
```

File: doc/README.md
Comment (charlesjhongc) [reply doc/README.md:19]: Yeah I think how strategy contracts are integrated should be at least mentioned here. Will fix.

File: doc/README.md
Comment (charlesjhongc) [reply doc/README.md:42]: Then we still need a way to describe those contracts as a whole. `Miscellaneous Contracts` and `Trading Contracts` can be subsets of that.

**Comment by NIC619** [comment]
I think one core proposition of Tokenlon is to served as a convenient and safe entry for user to access different protocols on-chain.
- Convenient because users only need to approve Tokenlon once (one for each token)
- Safe because Tokenlon can guard user against rogue contracts (user don't have to worry about his approvals for rouge contracts)

User can use different "strategy" to do different things conveniently and safely. Strategy can be simple or complex: it can be as simple as a dust sweeper, an airdrop tool, an AMM swap, a bridge to L2 to a complex Defi protocol combo.
To me "strategy" is not tied to "trading". I would prefer "strategy" over "executing" because "executing" seem too opaque to me personally.

File: doc/README.md
Comment (charlesjhongc) [reply doc/README.md]: Currently protocol doc is in `README.md` cause it can be displayed at the beginning and parts for permit will be added in this place once feature is merged. `advanced` is a not clear for me what should be inside if I'm not familiar with codebas.  `Peripherals` will have a sub cat just like `strategy`.

**Comment by charlesjhongc** [comment]
Adjusting `strategy` naming in f9a5173

**Review by changwu-tw — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**
